File: browser-cli-extension/src/background/dialog_actions.js
import {
  dialogOverrideJs,
  dialogRemoveJs,
} from '../page_runtime.js';
import { handleJavascriptDialog } from '../debugger.js';

export function createDialogHandlers(context) {
  return {
    async 'dialog-setup'(payload) {
      await context.getManagedTab(payload.tab_id);
      return await context.executeExpression(
        payload.tab_id,
        dialogOverrideJs({ action: payload.action, text: payload.text || null }),
      );
    },
    async dialog(payload) {
      await context.getManagedTab(payload.tab_id);
      await handleJavascriptDialog(payload.tab_id, {
        accept: !payload.dismiss,
        promptText: payload.text || null,
      });
      return { handled: true, accepted: !payload.dismiss, prompt_text: payload.text || null };
    },
    async 'dialog-remove'(payload) {
      await context.getManagedTab(payload.tab_id);
      return await context.executeExpression(payload.tab_id, dialogRemoveJs());
    },
  };
}

File: src/browser_cli/drivers/_extension/dialog_actions.py
from __future__ import annotations


class ExtensionDriverDialogActionsMixin:
    async def setup_dialog_handler(
        self,
        page_id: str,
        *,
        default_action: str = "accept",
        default_prompt_text: str | None = None,
    ) -> dict[str, object]:
        payload = await self._page_command(
            page_id,
            "dialog-setup",
            {"action": default_action, "text": default_prompt_text},
        )
        return {"page_id": page_id, **payload}

    async def handle_dialog(
        self,
        page_id: str,
        *,
        accept: bool,
        prompt_text: str | None = None,
    ) -> dict[str, object]:
        payload = await self._page_command(
            page_id,
            "dialog",
            {"dismiss": not accept, "text": prompt_text},
        )
        return {"page_id": page_id, **payload}

    async def remove_dialog_handler(self, page_id: str) -> dict[str, object]:
        payload = await self._page_command(page_id, "dialog-remove", {})
        return {"page_id": page_id, **payload}
